content/articles/jsdoc-documentation/index.md

```diff
@@ -0,0 +1,289 @@
+
+### Goal
+By the end of this tutorial, the reader should have learned how to initialize JSDocs in a JavaScript program and use it in a real-life programming scenario. We will write code snippets that are documented using JSDocs to demonstrate the concept.
+
+### Insight
+If you take a look at the function below, it is easy to tell what the function does as it speaks for itself. From the function name to its parameters, we can tell that this function calculates the area of a rectangle by taking the length and width as parameters.
+
+```Javascript
+function calculateArea(length, width, area) {
+	area = length * width;
+	return area;
+}
+```
+
+If we change the function and parameter names, this function could mean something else. It turns out that we could use the JSDocs documentation API to help us describe the function better than it speaks for itself. With the help of JSDocs, you add a line of comment starting with an asterisk `*` before the function name to document it:
+
+```Javascript
+/**
+* function to  calculate the area of a rectangle
+*/
+function calculateArea(length, width, area){
+    area = length * width;
+    return area;
+}}
+```
+
+### JSDocs Annotations
+Writing the function name alone is not enough. With JSDocs annotation, we can make things much interesting by documenting parameters as well. For each parameter, we take note of its type name and the description of what it does in the code. The syntax for a function parameter is shown below.
+
+```Javascript
+/**
+ *
+ * @param {parameter type} parameter name - parameter description
+ *
+ */
+```
+
+- Type: Parameter type may be a string, integer, array, floating-point, etc.
+- Name: Every parameter must have a name for referencing it in the code and when the function is called.
+- Description: An explanation of what the parameter is about.
+
+### Project Setup
+We have a glimpse of how to use JSDocs documentation API. To understand more, create a folder and name it `project`. Head over to your terminal and run the following commands.
+
+```bash
+npm init -y
+npm i- dev jsdocs
+```
+
+To use JSDocs, we need a config file. So in the root folder of the application, create a file named `jsodc.json` and paste the following snippet.
+
+```Javascript
+{
+	"source": {
+		"include": ["source"],
+		"includePattern": ".js$",
+		"excludePattern": "(node_modules/|docs)"
+	},
+	"plugins": ["plugins/markdown"],
+	"templates": {
+		"cleverLinks": true,
+		"monospaceLinks": true
+	},
+	"opts": {
+		"recurse": true,
+		"destination": "./documentation/"
+	}
+}
+```
+
+In the root folder, create a new folder named `source` and in the folder, add a new file named `index.js`. Here is where we are going to write the code for which its documentation is to be generated.
+
+### Creating the Kickoff Script
+At the snippets below in the Script Object in the `package.json` file:
```

Comment:
I don't understand this sentence. Please elaborate it.

Reply:
```json 
"scripts": {
    "doc": "jsdoc -c jsdoc.json"
  },
```

Reply:
@kayere Thank you for your observation, It is the script that will trigger the module to generate the documentation

Reply:
I still don't get the meaning of the sentence.

Reply:
If you look at the  `package.json` file I have to specify a command that I will use to start generating the documentation, just like you can use `nodemon start` to start a server in a nodejs application.

Reply:
So you meant `Add the snippet below in the Script Object in the `package.json` file:` instead of  `At the snippets below in the Script Object in the `package.json` file:`

Reply:
Yeah sure.